src/store/reducers/reducer.js

```diff
@@ -29,16 +29,11 @@ export const reducer = (state, action) => {
       };
     }
     case UPDATE_PRODUCT: {
-      const newProducts = state.product.map(item =>{
-        if(item.id === action.payload.id)
-          return action.payload
-        else
-          return item
-      })
-      return {
-        ...state,
-        product: newProducts
-      };
+     state.product[action.payload.index] = action.payload.newProduct
+     return {
```

Comment:
check the index of products by searching in the state.product,should not come from payload